src/responder/kcm/kcmsrv_ccache_secdb.c

```diff
@@ -469,23 +453,18 @@ static errno_t secdb_get_cc(TALLOC_CTX *mem_ctx,
         goto done;
     }
 
-    ret = sec_get(tmp_ctx, sreq, &ccbuf, &datatype);
+    ret = sec_get(tmp_ctx, sreq, &ccbuf);
     if (ret != EOK) {
         DEBUG(SSSDBG_OP_FAILURE,
               "Cannot get the secret [%d][%s]\n", ret, sss_strerror(ret));
         goto done;
     }
 
-    if (strcmp(datatype, "binary") == 0) {
-        ret = sec_kv_to_ccache_binary(tmp_ctx, secdb_key, ccbuf, client, &cc);
-    } else {
-        ret = sec_kv_to_ccache_json(tmp_ctx, secdb_key,
-                                    (const char *)sss_iobuf_get_data(ccbuf),
-                                    client, &cc);
-    }
+    ret = sec_kv_to_ccache_binary(tmp_ctx, secdb_key, ccbuf, client, &cc);
     if (ret != EOK) {
-        DEBUG(SSSDBG_OP_FAILURE, "Cannot convert %s data to ccache "
-              "[%d]: %s\n", datatype, ret, sss_strerror(ret));
+        DEBUG(SSSDBG_OP_FAILURE, "Cannot convert data to ccache [%d]: %s, "
+              "deleting this entry\n", ret, sss_strerror(ret));
+        sss_sec_delete(sreq);
```

Comment:
@pbrezina, is it ok to delete malformed ccache here?
My idea is to purge JSON-formatted ccaches is there is any in the wild.

Reply:
Fedora supports upgrade only over two release so we are fine here. The secrets file may still contain some ccache in old format but it will be most probably expired anyway so purging is fine. We need to get rid of it and make sure kcm will return non-failing error code so it won't break things.

Reply:
`ret = ENOENT`?
Or retry to try to get next ccache?

Reply:
Probably.